lightly/api/api_workflow_download_dataset.py

```diff
@@ -299,4 +305,53 @@ def export_filenames_by_tag_name(
 
         """
         tag = self.get_tag_by_name(tag_name)
-        return self.export_filenames_by_tag_id(tag.id)
\ No newline at end of file
+        return self.export_filenames_by_tag_id(tag.id)    
+
+
+    def export_filenames_and_read_urls_by_tag_id(
+        self,
+        tag_id: str,
+    ) -> List[Dict]:
+        """Export the samples filenames to map with their readURL.
+
+        Args:
+            tag_id:
+                Id of the tag which should exported.
+
+        Returns:
+            A list of mappings of the samples filenames and readURLs within a certain tag.
+
+        """
+        mappings = paginate_endpoint(
+            self._tags_api.export_tag_to_basic_filenames_and_read_urls,
+            page_size=10000,
+            dataset_id=self.dataset_id,
+            tag_id=tag_id
+        )
+        return mappings
+
+    def export_filenames_and_read_urls_by_tag_name(
+        self,
+        tag_name: str,
+    ) -> List[Dict]:
```

Comment:
This should be a List[ReadURLMapping]. Please be aware that a `ReadURLMapping` is a class, not a dict, despite the name.